modules/download/DownloadQuery.js

```diff
@@ -23,9 +23,14 @@ class DownloadQuery extends Query {
 
         const PROGRESS_TEMPLATE = '[download] %(progress._percent_str)s %(progress._speed_str)s %(progress._eta_str)s %(progress)j';
 
-        let tempFolderName = "temp[" + this.identifier + "]";
-        let tempFolderPath = this.environment.settings.downloadPath + "/" + tempFolderName;
-        let output = path.join(tempFolderPath, Utils.resolvePlaylistPlaceholders(this.environment.settings.nameFormat, this.playlistMeta));
+        let VideoDataFolderName = this.video.getFilename() + "_Data";
+        if(this.environment.settings.avoidFailingToSaveDuplicateFileName)
+            VideoDataFolderName += "[" + this.video.identifier + "]";
+        else
+            VideoDataFolderName += "[" + this.video.GetURLID() + "]";
```

Comment:
since this is the default case, we also do not have to create any temp folder. lets just use the previous logic here

Reply:
what do you mean by default case?

Reply:
The one using 'this.video.identifier' will never fail to save a duplicate file name but Resumability won't work, while the one using 'this.video.GetURLID()' has an edge case of failing to save a duplicate name (like when the user is downloading the same video more than once) but it lets you use Resumability.